### DISCUSSION TITLE
Best way to limit window size?

### DISCUSSION THREAD

**Discussion by dhardy** [rust-windowing/winit#3184] [General]
Kas uses an algorithm to determine window size from its contents. As you can guess, [someone](https://github.com/kas-gui/kas/issues/415) managed to break this by stuffing too much content in the window, thus trying to open a very large window (succeeds) and create a backing texture (fails somewhere around 16000 pixels, presumably a 16k texture size limit).

What's the correct approach to limit this to something sane?

- A hard-coded maximum?
- Query the maximum supported texture size? Applying this is hard to do since Wayland does not really support specification of window size in physical pixels.
- Limit to the maximum monitor size? This isn't possible to do correctly: #3183

**Comment by kchibisov** [comment]
There's a https://wayland.app/protocols/xdg-shell#xdg_toplevel:event:configure_bounds created just for that, it's not supported by all the compositors, but that's exactly why it exists.

**Comment by kchibisov** [comment]
>Limit to the maximum monitor size? This isn't possible to do correctly: https://github.com/rust-windowing/winit/issues/3183

This is possible to do correctly by limiting `PhysicalSize` if you take the monitor resolution, because it's physical and use it. Since what you want to limit is buffer sizes, not the window sizes, given that you're getting a texture error. Also, some drivers expose the maximum size of the texture, so you can limit by that.

**Reply by dhardy** [reply]
> by limiting PhysicalSize

Constructing a Wayland window using `PhysicalSize` does not work correctly: the result is too big. Does this mean I should always resize afterwards?

**Reply by kchibisov** [reply]
Hm, I think I can change that to actually work with modern compositors.

**Reply by dhardy** [reply]
That's #2275 BTW.

**Reply by kchibisov** [reply]
yeah, but I can't really fix it for all setups, only for new protocol.

**Comment by kchibisov** [comment]
https://github.com/rust-windowing/winit/issues/3185 for window sizes, but generally, you could have a different texture limit.
